crates/sparrow-runtime/src/prepare/column_behavior.rs

```diff
@@ -24,8 +25,15 @@ pub enum ColumnBehavior {
     /// Cast the given column to the given data type.
```

Comment:
Do we need to use / change the column behavior? I think we could just use the `prepare_batch` method and avoid that. Alternatively, it seems like we could have a case here that called into the `numeric_to_timestamp` and at least shared that case (rather than adding more complexity to column behaviors).